Source/PartnerCenter.CustomerPortal/BusinessLogic/Commerce/PaymentGateways/PayUGateway.cs

```diff
@@ -0,0 +1,437 @@
+            return paymentConfig.WebExperienceProfileId;
+        }
+
+        /// <summary>
+        /// Creates a payment transaction and returns the PayUMoney generated payment URL. 
+        /// </summary>
+        /// <param name="returnUrl">The redirect url for PayUMoney callback to web store portal.</param>                
+        /// <param name="order">The order details for which payment needs to be made.</param>        
+        /// <returns>Payment URL from PayUMoney.</returns>
+        public async Task<string> GeneratePaymentUriAsync(string returnUrl, OrderViewModel order)
+        {
+            returnUrl.AssertNotEmpty(nameof(returnUrl));
+            order.AssertNotNull(nameof(order));
+            RemotePost myremotepost = await this.PrepareRemotePost(order, returnUrl);
+            return myremotepost.Post();
+        }
+
+        /// <summary>
+        /// Executes a PayU payment.
+        /// </summary>
+        /// <returns>Capture string id.</returns>
+        public async Task<string> ExecutePaymentAsync()
+        {
+            try
+            {
+                TransactionStatusResponse paymentResponse = await ApiCalls.GetPaymentStatus(this.paymentId);
+                if (paymentResponse != null && paymentResponse.Result.Count > 0 && paymentResponse.Result[0].Status.Equals(Constant.MoneyWithPayU))
+                {
+                    return paymentResponse.Result[0].Amount.ToString();
+                }
+            }
+            catch (PartnerDomainException ex)
+            {
+                this.ParsePayUException(ex);
+            }
+
+            return await Task.FromResult(string.Empty);
+        }
+
+        /// <summary>
+        /// Finalizes an authorized payment with PayU.
+        /// </summary>
+        /// <param name="authorizationCode">The authorization code for the payment to capture.</param>
+        /// <returns>A task.</returns>
+        public async Task CaptureAsync(string authorizationCode)
+        {
+            ////PayU api not provided
+            await Task.FromResult(string.Empty);
+        }
+
+        /// <summary>
+        /// Voids an authorized payment with PayUMoney.
+        /// </summary>
+        /// <param name="authorizationCode">The authorization code for the payment to void.</param>
+        /// <returns>a Task</returns>
+        public async Task VoidAsync(string authorizationCode)
+        {
+            authorizationCode.AssertNotEmpty(nameof(authorizationCode));
+
+            // given the authorizationId string... Lookup the authorization to void it. 
+            try
+            {
+                RefundResponse refundResponse = await ApiCalls.RefundPayment(this.payerId, authorizationCode);
+                if (refundResponse.Status != 0 || !refundResponse.Message.Equals("Refund Initiated"))
+                {
+                    throw new Exception("Error in refund");
+                }
+            }
+            catch (PartnerDomainException ex)
```

Comment:
>PartnerDomainException [](start = 19, length = 22)

You are throwing a regular Exception but catching PartnerDomainException.

Reply:
done.

---
In reply to: [183243743](https://github.com/PartnerCenterSamples/Reseller-Web-Application/pull/61#discussion_r183243743) [](ancestors = 183243743)